config.go

```diff
@@ -0,0 +1,217 @@
+/*
+  Copyright 2011 Alec Thomas
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"gopkg.in/alecthomas/kingpin.v2"
+	"io"
+	"os"
+	"sync"
+)
+
+var (
+	instance *config
+	once     sync.Once
+)
+
+type config struct {
+	FGColors   map[Priority]string
+	BGColors   map[Priority]string
+	Priority   string
+	Graft      string
+	File       string
+	LegacyFile string
+	Order      string
+	ConfigFile string
+}
+
+// Map for priority level to ANSI colour
+var colourPriorityMap = map[Priority]string{
+	VERYHIGH: BRIGHT + FGRED,
+	HIGH:     BRIGHT + FGYELLOW,
+	MEDIUM:   FGWHITE,
+	LOW:      FGCYAN,
+	VERYLOW:  FGBLUE,
+}
+
+func GetConfigInstance() *config {
+	once.Do(func() {
```

Comment:
Alternatively, just have a single global config instance. For a simple project like this, that is not such a big deal. That would also let you map the flag definitions directly to configuration fields, rather than having to manually copy them.